Gems/Atom/Feature/Common/Code/Source/SkinnedMesh/SkinnedMeshOutputStreamManager.cpp

```diff
@@ -72,11 +74,36 @@ namespace AZ
             creator.End(m_bufferAsset);
         }
 
+        AZ_CVAR(
+            int,
+            r_skinnedMeshInstanceMemoryPoolSize,
+            256,
+            nullptr,
+            AZ::ConsoleFunctorFlags::NeedsReload,
+            "The amount of memory in Mb available for all actor skinning data. Note that this must only be set once at application startup"
+        );
+
         void SkinnedMeshOutputStreamManager::Init()
         {
+        }
+
+        void SkinnedMeshOutputStreamManager::EnsureInit() const
+        {
+            const_cast<SkinnedMeshOutputStreamManager*>(this)->EnsureInit();
+        }
+
+        void SkinnedMeshOutputStreamManager::EnsureInit()
+        {
+            if (!m_needsInit)
+            {
+                return;
+            }
+            m_needsInit = false;
+
             // 256mb supports roughly 42 character instances at 100,000 vertices per character x 64 bytes per vertex (12 byte position + 12 byte previous frame position + 12 byte normal + 16 byte tangent + 12 byte bitangent)
             // This includes only the output of the skinning compute shader, not the input buffers or bone transforms
-            m_sizeInBytes = 256u * (1024u * 1024u);
+            const AZ::u64 sizeInMb = r_skinnedMeshInstanceMemoryPoolSize;
+            m_sizeInBytes = AZ::GetMax(sizeInMb, 256ull) * (1024u * 1024u);
```

Comment:
I think 256MB might be a little large for mobile platforms, lets set the minimum to 8MB. Also can we move the minimum to a named constant at the top of the file.

Reply:
The only reason I can think of to have a minimum at all is to avoid attempting to allocate a zero size buffer if someone sets the value to 0, which would potentially result in validation errors in the RHI but we could probably handle that with an early-out in the EnsureInit function. You'd just need to separate CreateBufferAsset into CreateAllocator and CreateBufferAsset so that we can get a valid zero-size allocator that can return an invalid address if someone tries to allocate from it, the same as it would if someone tries to allocate from beyond the capacity of any other size buffer (assuming the free list allocator handles a size of 0).

256mb is just a default, not something that is required. The memory is only needed/used if someone actually tries to create a skinned mesh.